src/common/dns_resolve.cc

```diff
@@ -314,7 +314,7 @@ int DNSResolver::resolve_srv_hosts(CephContext *cct, const string& service_name,
   }
 #endif
   if (len < 0) {
-    lderr(cct) << "res_search() failed" << dendl;
+    lderr(cct) << "res_search() failed" << query_str << dendl;
```

Comment:
Your part of the fix is good. But please check again:

`#ifdef HAVE_RES_NQUERY`

then it is not calling res_search(). So the text "res_search() failed" is not correct.